packages/verified-fetch/src/verified-fetch.ts

```diff
@@ -289,6 +294,11 @@ export class VerifiedFetch {
       if (['ERR_NO_PROP', 'ERR_NO_TERMINAL_ELEMENT'].includes(err.code)) {
         return notFoundResponse(resource)
       }
+      if (err.code === 'ERR_NO_ROUTERS_AVAILABLE') {
+        // need to retry without sessions
+        // TODO: handle this better
+        return this.handleDagCbor({ resource, cid, path, accept, session: false, cacheKey, options })
+      }
```

Comment:
AFAICT, there is no way to check `this.helia.routing` for routers length? feels like this should be handled in helia session management code